src/examples/libpmemobj/cpp_map/ctree_map_persistent.hpp

```diff
@@ -356,9 +356,18 @@ class ctree_map_p {
 	 * Find critical bit.
 	 */
 	static int
-	find_crit_bit(key_type lhs, key_type rhs)
+	find_crit_bit(uint64_t lhs, uint64_t rhs)
 	{
+#ifndef _WIN32
 		return 64 - __builtin_clzll(lhs ^ rhs) - 1;
+#else
+		DWORD lz = 0;
+
+		if (BitScanReverse64(&lz, lhs ^ rhs))
```

Comment:
IMHO we should create a tiny common header-only library for the often repeating bolierplate stuff.